tests/failover.go

```diff
@@ -445,3 +446,238 @@ func (oa *operatorActions) GetNodeMap(info *TidbClusterConfig, component string)
 
 	return nodeMap, nil
 }
+
+func (oa *operatorActions) CheckOneEtcdDownOrDie(operatorConfig *OperatorConfig, clusters []*TidbClusterConfig, faultNode string) {
+	KeepOrDie(3*time.Second, 10*time.Minute, func() error {
+		err := oa.CheckK8sAvailable(nil, nil)
+		if err != nil {
+			return err
+		}
+		glog.Infof("k8s cluster is available.")
+		err = oa.CheckOperatorAvailable(operatorConfig)
+		if err != nil {
+			return err
+		}
+		glog.Infof("tidb operator is available.")
+		err = oa.CheckTidbClustersAvailable(clusters)
+		if err != nil {
+			return err
+		}
+		glog.Infof("all clusters is available")
+		return nil
+	})
+}
+
+func (oa *operatorActions) CheckOneApiserverDownOrDie(operatorConfig *OperatorConfig, clusters []*TidbClusterConfig, faultNode string) {
+	affectedPods := map[string]*corev1.Pod{}
+	apiserverPod, err := GetApiserverPod(oa.kubeCli, faultNode)
+	if err != nil {
+		panic(fmt.Errorf("can't find apiserver in node:%s", faultNode))
+	}
+	if apiserverPod != nil {
+		affectedPods[apiserverPod.GetName()] = apiserverPod
+	}
+	controllerPod, err := GetControllerManagerPod(oa.kubeCli, faultNode)
+	if err != nil {
+		glog.Infof("can't find controllerManager in node:%s", faultNode)
+	}
+	if controllerPod != nil {
+		affectedPods[controllerPod.GetName()] = controllerPod
+	}
+	schedulerPod, err := GetSchedulerPod(oa.kubeCli, faultNode)
+	if err != nil {
+		glog.Infof("can't find scheduler in node:%s", faultNode)
+	}
+	if schedulerPod != nil {
+		affectedPods[schedulerPod.GetName()] = schedulerPod
+	}
+	dnsPod, err := GetDnsPod(oa.kubeCli, faultNode)
+	if err != nil {
+		panic(fmt.Errorf("can't find controller-manager in node:%s", faultNode))
+	}
+	if dnsPod != nil {
+		affectedPods[dnsPod.GetName()] = dnsPod
+	}
+	KeepOrDie(3*time.Second, 10*time.Minute, func() error {
+		err := oa.CheckK8sAvailable(nil, affectedPods)
+		if err != nil {
+			return err
+		}
+		glog.Infof("k8s cluster is available.")
+		err = oa.CheckOperatorAvailable(operatorConfig)
+		if err != nil {
+			return err
+		}
+		glog.Infof("tidb operator is available.")
+		err = oa.CheckTidbClustersAvailable(clusters)
+		if err != nil {
+			return err
+		}
+		glog.Infof("all clusters is available")
+		return nil
+	})
+}
+
+func (oa *operatorActions) CheckK8sAvailable(excludeNodes map[string]*corev1.Node, excludePods map[string]*corev1.Pod) error {
+	return wait.Poll(3*time.Second, time.Minute, func() (bool, error) {
```

Comment:
use the default interval

Reply:
default interval is too long for the case